cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metrics.store.timeseries;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.metrics.data.EntityTable;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Helper for serde of Fact into columnar format.
+ */
+public class FactCodec {
+  private static final Logger LOG = LoggerFactory.getLogger(FactCodec.class);
+  // encoding types
+  private static final String TYPE_MEASURE_NAME = "measureName";
+  private static final String TYPE_TAGS_GROUP = "tagsGroup";
+
+  private final EntityTable entityTable;
+
+  private final int resolution;
+  private final int rollTimebaseInterval;
+  // Cache for delta values.
+  private final byte[][] deltaCache;
+
+  public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterval) {
+    this.entityTable = entityTable;
+    this.resolution = resolution;
+    this.rollTimebaseInterval = rollTimebaseInterval;
+    this.deltaCache = createDeltaCache(rollTimebaseInterval);
+  }
+
+  public byte[] createRowKey(List<TagValue> tagValues, String measureName, long ts) {
```

Comment:
Added javadoc. One for get and write op, another for scan. At the moment they behave the same, but from the contract standpoint - these may be different.